web-client/src/ducks/auth/phone/types.ts

```diff
@@ -0,0 +1,18 @@
+import createActionTypeFactory from 'src/store/utils/createActionTypeFactory';
+
+const { asyncType } = createActionTypeFactory('FIREBASE');
+
+export const FIREBASE_PHONE_TRIGGER = asyncType('PHONE_TRIGGER');
+
+export const FIREBASE_PHONE_VERIFY = asyncType('PHONE_VERIFY');
+
+export interface IPhoneNumberAuth {
+  phoneNumber: string;
+  currentUser?: firebase.User;
+  // eslint-disable-next-line @typescript-eslint/camelcase
+  recaptchaVerifier: firebase.auth.RecaptchaVerifier_Instance;
+}
+
+export interface IOTPAuth {
```

Comment:
That makes sense. I didn't notice I was using them outside /phone

Reply:
what I meant was that the type should be defined in the same file as the file in which the action creator is defined.
you have defined all action creator types in the main types of the redux module. 
This still has the same issue I mentioned above, but now also has the issue of too much code in one place

Reply:
So, you mean doing this in actions.ts?

Reply:
I've been thinking about these types, I think we should try to abstract them into a directory in our project. 
Except for Interfaces that are very specific, such as an interface to represent a component's props, we should declare these interfaces in a more common space because we never know where we're going to need them.

Reply:
yes I mean in actions.ts
but the thing is interfaces are mostly going to be needed when we are interacting with an action creator.
and the interface for interacting with the action creator may not be the interface when interacting with the actual function as seen in the verify OTP action creator.
so I think defining them along with the actions creators in the actions.ts would be the best solution

Reply:
We had a discussion over this in PM.
We are considering various options on how to do this. but we decided to raise this as a separate issue for now